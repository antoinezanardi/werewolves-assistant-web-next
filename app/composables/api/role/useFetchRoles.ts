import type { Role } from "~/composables/api/role/types/role.class";
import { useWerewolvesAssistantApi } from "~/composables/api/useWerewolvesAssistantApi";

type UseFetchRoles = {
  fetchRoles: () => Promise<ReturnType<typeof $fetch<Role[]>> | null>;
};

function useFetchRoles(): UseFetchRoles {
  const { fetchWerewolvesAssistantApi } = useWerewolvesAssistantApi();

  async function fetchRoles(): Promise<ReturnType<typeof $fetch<Role[]>> | null> {
    try {
      return await fetchWerewolvesAssistantApi<Role[]>(`/roles`, { method: "GET" });
    } catch {
      return null;
    }
  }
  return { fetchRoles };
}

export { useFetchRoles };