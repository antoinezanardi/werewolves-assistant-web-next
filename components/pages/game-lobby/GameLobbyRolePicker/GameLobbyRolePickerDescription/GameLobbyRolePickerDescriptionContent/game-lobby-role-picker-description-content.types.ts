import type { Role } from "~/composables/api/role/types/role.class";

type GameLobbyRolePickerDescriptionContentProps = {
  pickedRole: Role
};

export type { GameLobbyRolePickerDescriptionContentProps };