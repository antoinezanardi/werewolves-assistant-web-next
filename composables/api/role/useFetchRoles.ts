import type { UseFetchOptions } from "#app";

import { DEFAULT_USE_FETCH_OPTIONS } from "~/composables/api/api.constants";
import type { Role } from "~/composables/api/role/types/role.types";
import { removeTrailingSlashes } from "~/utils/url.utils";

type UseFetchRoles = {
  fetchRoles: (options?: UseFetchOptions<Role[]>) => Promise<ReturnType<typeof useFetch<Role[] | null>>>;
};

function useFetchRoles(): UseFetchRoles {
  const config = useRuntimeConfig();
  const baseUrl = removeTrailingSlashes(config.public.werewolvesAssistantApi.baseUrl);

  async function fetchRoles(options: UseFetchOptions<Role[]> = {}): Promise<ReturnType<typeof useFetch<Role[] | null>>> {
    return useFetch<Role[]>(`${baseUrl}/roles`, {
      method: "GET",
      ...DEFAULT_USE_FETCH_OPTIONS,
      ...options,
    });
  }
  return { fetchRoles };
}

export { useFetchRoles };