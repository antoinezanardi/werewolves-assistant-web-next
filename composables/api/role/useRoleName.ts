import type { RoleName } from "~/composables/api/role/types/role.types";

type UseRoleName = {
  getRoleNameLabel: (roleName: RoleName) => string;
};

function useRoleName(): UseRoleName {
  const { t } = useI18n();

  function getRoleNameLabel(roleName: RoleName): string {
    return t(`shared.role.name.${roleName}`);
  }
  return { getRoleNameLabel };
}

export { useRoleName };