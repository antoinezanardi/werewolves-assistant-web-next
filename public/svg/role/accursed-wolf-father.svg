<svg xmlns="http://www.w3.org/2000/svg" width="1em" height="1em" viewBox="0 0 64 64"><path fill="#728389" d="m7.9 28.2l18.9-13S16 .1 10.6 2.2c-2.6 1-4.5 21.9-2.7 26"/><path fill="#3e4347" d="m8.7 29.8l7.5-3.6S20.6 14 11.1 5.3C9.2 6.9 6.9 26.9 8.7 29.8"/><path fill="#728389" d="m56.1 28.2l-18.9-13S48 .1 53.4 2.2c2.6 1 4.5 21.9 2.7 26"/><path fill="#3e4347" d="m55.3 29.8l-7.5-3.6S43.4 14 52.9 5.3c1.9 1.6 4.2 21.6 2.4 24.5"/><path fill="#728389" d="M32 9.7C18.1 9.7 2 25.5 2 36.9C2 46.1 27.9 62 32 62c4.1 0 30-15.9 30-25.1C62 25.5 45.9 9.7 32 9.7"/><path fill="#dae5ef" d="M4 36.6c.4 9.2 24 21.7 28 21.7s27.6-12.5 28-21.7C32.5 9.1 52.9 54.5 32 54.5S31.5 9.1 4 36.6"/><path fill="#3e4347" d="M40.5 48.9c-1.7-1.1-6.9-1.2-8.5-1.2c-1.6 0-6.8.1-8.5 1.2c-1.2.8-.3 2.7 2.9 4.4c2 1 4 1.4 5.5 1.4c1.6 0 3.5-.3 5.5-1.4c3.3-1.7 4.3-3.6 3.1-4.4"/><g fill="#728389"><path d="M11.9 13.9c2.2 1.3 8.2 7.6 3.5 11.2s-7.4 0-6.8-.6c3.9-3.6 3.3-10.6 3.3-10.6"/><path d="M10 16.4c2.3 1 8.8 6 4.3 9.9c-4.5 3.9-7.4.9-6.8.2c3.6-3.9 2.5-10.1 2.5-10.1m42.1-2.5c-2.2 1.3-8.2 7.6-3.5 11.2s7.4 0 6.8-.6c-3.9-3.6-3.3-10.6-3.3-10.6"/><path d="M54 16.4c-2.3 1-8.8 6-4.3 9.9c4.5 3.9 7.4.9 6.8.2c-3.6-3.9-2.5-10.1-2.5-10.1"/></g><g fill="#dae5ef"><path d="M22.2 19.4c2.1 3.6 6.4 17.5-1.7 20s-9.6-5-8.3-5.5c6.9-3.3 10-14.5 10-14.5"/><path d="M18.4 21.9c2.4 3.1 7.9 15.4.1 18.4c-7.9 3.1-10-3.6-8.8-4.2c6.6-3.6 8.7-14.2 8.7-14.2"/></g><path fill="#42ade2" d="M23.2 33.1c0 3.3-2.7 6-6 6s-6-2.7-6-6s2.7-4 6-4z"/><path fill="#3e4347" d="M21 33.1c0 2.1-1.7 3.8-3.8 3.8c-2.1 0-3.8-1.7-3.8-3.8c0-2.1 1.7-2.8 3.8-2.8c2.1-.1 3.8.6 3.8 2.8"/><g fill="#dae5ef"><path d="M24.1 32.7c-.2-.5-.6-.9-.9-1.2c-.7-.7-1.5-1.3-2.2-1.8c-1.5-1-3.1-1.8-4.6-2.6c-1.5-.8-3-1.6-4.3-2.6c-1.3-1-2.5-2.2-3.4-3.9c-.1.9.1 1.8.3 2.6c-1.3-1.2-2.5-2.6-3.4-4.5c-.1 1.3.2 2.6.6 3.8c-1.1-1.2-2.1-2.6-2.9-4.3c-.1 2.5.9 5 2.4 7s3.5 3.4 5.5 4.5s4.1 2 6.2 2.7c1 .4 2 .8 3 1.3c.9.5 1.8 1 2.4 2.1zm17.7-13.3c-2.1 3.6-6.4 17.5 1.7 20s9.6-5 8.3-5.5c-6.9-3.3-10-14.5-10-14.5"/><path d="M45.6 21.9c-2.4 3.1-7.9 15.4-.1 18.4c7.9 3.1 10-3.6 8.8-4.2c-6.6-3.6-8.7-14.2-8.7-14.2"/></g><path fill="#42ade2" d="M40.8 33.1c0 3.3 2.7 6 6 6s6-2.7 6-6s-2.7-4-6-4z"/><path fill="#3e4347" d="M43 33.1c0 2.1 1.7 3.8 3.8 3.8c2.1 0 3.8-1.7 3.8-3.8c0-2.1-1.7-2.8-3.8-2.8c-2.1-.1-3.8.6-3.8 2.8"/><path fill="#dae5ef" d="M39.9 32.7c.2-.5.6-.9.9-1.2c.7-.7 1.5-1.3 2.2-1.8c1.5-1 3.1-1.8 4.6-2.6c1.5-.8 3-1.6 4.3-2.6c1.3-1 2.5-2.2 3.4-3.9c.1.9-.1 1.8-.3 2.6c1.3-1.2 2.5-2.6 3.4-4.5c.1 1.3-.2 2.6-.6 3.8c1.1-1.2 2.1-2.6 2.9-4.3c.1 2.5-.9 5-2.4 7s-3.5 3.4-5.5 4.5s-4.1 2-6.2 2.7c-1 .4-2 .8-3 1.3c-.9.5-1.8 1-2.4 2.1z"/><path fill="#728389" d="M38.6 56.9c0 1.2-4.1 2.3-6.6 2.3s-6.6-1.1-6.6-2.3s6.6-.7 6.6-.7s6.6-.4 6.6.7"/><g fill="#3e4347"><path d="M33 53.5v.4c0 .2 0 .4-.1.5c-.1.3-.2.7-.3 1c-.3.7-.8 1.3-1.4 1.7c-1.3.8-2.7.6-3.9.4c-2.4-.5-4.4-1.6-6.4-2.8c-1.9-1.3-3.7-2.7-5.3-4.3c-1.6-1.6-3-3.4-4.2-5.3c1.3 1.9 2.8 3.5 4.5 5c1.7 1.5 3.5 2.8 5.5 3.8c1.9 1.1 4 1.9 6.1 2.2c1 .1 2.1 0 2.7-.4c.6-.4.8-1.2.8-2.2z"/><path d="M33 53.4c0 1 .2 1.8.8 2.2c.6.5 1.7.5 2.7.4c2.1-.3 4.2-1.2 6.1-2.2c1.9-1.1 3.8-2.4 5.5-3.8c1.7-1.5 3.2-3.1 4.5-5c-1.1 1.9-2.6 3.7-4.2 5.3c-1.6 1.6-3.4 3.1-5.3 4.3c-1.9 1.2-4 2.3-6.4 2.8c-1.2.2-2.5.3-3.9-.4c-.7-.4-1.2-1-1.4-1.7c-.1-.3-.2-.7-.3-1c0-.2-.1-.3-.1-.5v-.4z"/></g></svg>