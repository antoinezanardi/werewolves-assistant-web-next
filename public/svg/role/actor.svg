<?xml version="1.0" encoding="iso-8859-1"?>
<!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
<svg height="800px" width="800px" version="1.1" id="Layer_1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" 
	 viewBox="0 0 512 512" xml:space="preserve">
<path style="fill:#FFDB6C;" d="M147.68,328.121c0,34.446,0,40.894,0,75.338c0,67.233,44.782,108.54,100.022,108.54h12.346
	c55.241,0,100.022-41.307,100.022-108.54c0-34.446,0-40.892,0-75.338H147.68z M237.125,392.619l-9.98,2.981
	c-22.109,6.605-45.386-5.962-51.992-28.071l9.979-2.981C207.242,357.943,230.52,370.511,237.125,392.619L237.125,392.619z
	 M280.607,395.6l-9.979-2.981l0,0c6.605-22.109,29.883-34.676,51.991-28.071l9.979,2.981
	C325.992,389.638,302.716,402.206,280.607,395.6z"/>
<path style="fill:#FFB04C;" d="M312.116,328.121c0,15.562,0,25.408,0,34.704c3.412,0.139,6.862,0.7,10.284,1.722l9.979,2.981
	c-3.312,11.085-10.817,19.77-20.264,24.845c0,3.371,0,7.028,0,11.085c0,59.884-35.527,99.2-82.327,107.08
	c5.743,0.967,11.657,1.461,17.695,1.461h12.346c55.241,0,100.022-41.307,100.022-108.54c0-34.446,0-40.892,0-75.338L312.116,328.121
	L312.116,328.121z"/>
<g>
	<path style="fill:#FD4DD7;" d="M257.616,480.518h-7.479c-21.658,0-41.323-9.476-53.953-26.001
		c-2.852-3.732-2.139-9.069,1.593-11.922c3.731-2.853,9.069-2.14,11.921,1.592c9.523,12.46,23.884,19.321,40.439,19.321h7.479
		c16.313,0,30.542-6.689,40.065-18.837c2.898-3.699,8.244-4.343,11.939-1.446c3.697,2.898,4.344,8.244,1.446,11.939
		C298.436,471.278,278.954,480.518,257.616,480.518z"/>
	<path style="fill:#FD4DD7;" d="M188.999,24.413c-8.887-3.66-16.691,6.942-10.613,14.388c19.335,23.686,30.422,54.261,29.091,87.214
		c-0.406,10.067-1.938,19.522-4.469,28.332v119.968h101.089v-118.35C291.625,89.815,253.229,50.869,188.999,24.413z"/>
</g>
<circle style="fill:#FFDB6C;" cx="162.21" cy="18.313" r="18.313"/>
<path style="fill:#FFB04C;" d="M168.445,1.099c-1.887-0.683-3.812-1.024-5.714-1.078c6.044,4.75,8.666,13.001,5.906,20.623
	c-2.759,7.622-10.055,12.283-17.738,12.065c1.496,1.176,3.193,2.147,5.08,2.829c9.51,3.443,20.011-1.476,23.452-10.987
	C182.872,15.041,177.956,4.54,168.445,1.099z"/>
<circle style="fill:#FFDB6C;" cx="25.3" cy="223.501" r="18.313"/>
<path style="fill:#FFB04C;" d="M31.533,206.282c-1.004-0.363-2.02-0.613-3.036-0.795c5.424,4.835,7.669,12.65,5.045,19.901
	c-3.079,8.507-11.804,13.319-20.417,11.782c1.691,1.507,3.682,2.734,5.942,3.552c9.51,3.443,20.011-1.476,23.452-10.986
	C45.963,220.225,41.044,209.725,31.533,206.282z"/>
<circle style="fill:#FFDB6C;" cx="486.7" cy="220.893" r="18.313"/>
<path style="fill:#FFB04C;" d="M476.862,222.047c3.443-9.51,13.943-14.428,23.452-10.987c0.867,0.314,1.683,0.705,2.471,1.131
	c-2.061-3.815-5.46-6.925-9.857-8.517c-9.51-3.443-20.011,1.476-23.452,10.987c-3.129,8.643,0.663,18.084,8.516,22.322
	C475.594,232.549,475.011,227.159,476.862,222.047z"/>
<path style="fill:#FC4C59;" d="M360.395,274.316H148.839c-6.528,0-11.82,5.292-11.82,11.82v31.298c0,6.528,5.292,11.82,11.82,11.82
	h211.556c6.528,0,11.82-5.292,11.82-11.82v-31.298C372.215,279.608,366.923,274.316,360.395,274.316z"/>
<path style="fill:#E83A64;" d="M360.395,274.316h-43.394c6.528,0,11.82,5.292,11.82,11.82v31.298c0,6.528-5.292,11.82-11.82,11.82
	h43.394c6.528,0,11.82-5.292,11.82-11.82v-31.298C372.215,279.608,366.923,274.316,360.395,274.316z"/>
<path style="fill:#E846AA;" d="M300.14,138.938c-6.993,4.082-13.523,8.88-19.496,14.29c-10.008,9.066-26.027,1.914-26.027-11.59l0,0
	c-9.805-51.996-35.637-87.18-77.432-112.706c-1.451,2.969-1.382,6.703,1.202,9.869c19.335,23.686,30.422,54.261,29.091,87.214
	c-0.406,10.067-1.938,19.522-4.469,28.332v105.642l101.089,0.719V155.963C302.988,150.073,301.661,144.408,300.14,138.938z"/>
<path style="fill:#7C54B8;" d="M157.192,274.316h97.716v-22.166c0-63.959-51.143-116.935-115.1-117.48
	c-45.988-0.392-85.859,25.955-105.011,64.431c-3.88,7.795,5.155,15.832,12.43,11.047c12.087-7.949,26.555-12.574,42.103-12.574
	C131.714,197.572,157.192,231.931,157.192,274.316z"/>
<path style="fill:#674799;" d="M157.192,274.316h49.017c0-54.591-27.208-99.088-85.087-99.581
	c-32.905-0.28-62.672,13.132-83.965,34.878c2.681,2.168,6.576,2.831,10.069,0.534c12.088-7.949,26.555-12.574,42.103-12.574
	C131.714,197.572,157.192,231.931,157.192,274.316z"/>
<path style="fill:#3CC247;" d="M352.123,274.316h-97.458v-22.166c0-63.959,51.143-116.935,115.1-117.48
	c45.988-0.392,85.859,25.955,105.011,64.431c3.88,7.795-5.155,15.832-12.43,11.047c-12.087-7.949-26.555-12.574-42.103-12.574
	C377.858,197.572,352.123,231.931,352.123,274.316z"/>
<path style="fill:#2DA85F;" d="M389.042,173.223c-58.402,0.498-94.221,43.573-94.221,101.093h57.303
	c0-42.385,25.735-76.743,68.12-76.743c15.549,0,30.017,4.625,42.103,12.574c3.995,2.627,8.515,1.385,11.141-1.556
	C452.163,186.556,422.194,172.941,389.042,173.223z"/>
<g>
	<circle style="fill:#FFDB6C;" cx="169.671" cy="301.789" r="8.278"/>
	<circle style="fill:#FFDB6C;" cx="202.941" cy="301.789" r="8.278"/>
	<circle style="fill:#FFDB6C;" cx="236.222" cy="301.789" r="8.278"/>
	<circle style="fill:#FFDB6C;" cx="269.504" cy="301.789" r="8.278"/>
	<circle style="fill:#FFDB6C;" cx="302.774" cy="301.789" r="8.278"/>
	<circle style="fill:#FFDB6C;" cx="336.055" cy="301.789" r="8.278"/>
</g>
</svg>