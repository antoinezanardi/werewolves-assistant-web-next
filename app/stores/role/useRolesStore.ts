import type { AsyncDataRequestStatus } from "nuxt/app";
import { defineStore } from "pinia";
import { ref } from "vue";
import type { GameAdditionalCardRecipientRoleName } from "~/composables/api/game/types/game-additional-card/game-additional-card.types";

import type { Role } from "~/composables/api/role/types/role.class";
import type { RoleName, RoleSide } from "~/composables/api/role/types/role.types";
import { useFetchRoles } from "~/composables/api/role/useFetchRoles";
import { StoreIds } from "~/stores/enums/store.enum";

const useRolesStore = defineStore(StoreIds.ROLES, () => {
  const { fetchRoles } = useFetchRoles();

  const fetchingRoleStatus = ref<AsyncDataRequestStatus>("idle");
  const roles = ref<Role[] | null>(null);

  async function fetchAndSetRoles(): Promise<void> {
    fetchingRoleStatus.value = "pending";
    roles.value = await fetchRoles();
    fetchingRoleStatus.value = "success";
  }

  function getRoleWithNameInRoles(roleName: RoleName): Role | undefined {
    return roles.value?.find(role => role.name === roleName);
  }

  function getRolesForRecipientRoleName(recipientRoleName: GameAdditionalCardRecipientRoleName): Role[] {
    return roles.value?.filter(role => role.additionalCardsEligibleRecipients?.includes(recipientRoleName)) ?? [];
  }

  function getRoleSideForRoleName(roleName: RoleName): RoleSide | undefined {
    return getRoleWithNameInRoles(roleName)?.side;
  }
  return {
    roles,
    fetchingRoleStatus,
    fetchAndSetRoles,
    getRoleWithNameInRoles,
    getRolesForRecipientRoleName,
    getRoleSideForRoleName,
  };
});

export { useRolesStore };