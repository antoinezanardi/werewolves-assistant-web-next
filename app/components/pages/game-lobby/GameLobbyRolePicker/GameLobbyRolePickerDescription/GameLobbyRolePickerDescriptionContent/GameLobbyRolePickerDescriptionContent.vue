<template>
  <div
    id="game-lobby-role-picker-description-content"
    class="flex flex-col gap-4 items-center w-full"
  >
    <h3 id="role-name">
      {{ roleName }}
    </h3>

    <RoleSideBadge
      id="role-side-badge"
      :role-side="pickedRole.side"
      size="large"
    />

    <div class="flex gap-2 items-center justify-between w-full">
      <RoleOriginBadge
        id="role-origin-badge"
        :role-origin="pickedRole.origin"
      />

      <RoleTypeBadge
        id="role-type-badge"
        class="role-type-badge"
        :role-type="pickedRole.type"
      />
    </div>

    <GameLobbyRolePickerDescriptionBoundaries
      id="game-lobby-role-picker-description-boundaries"
      class="w-full"
      :picked-role="pickedRole"
    />

    <PrimeVueDivider class="!my-2"/>

    <RoleDescriptionLines
      id="role-description-lines"
      :role="pickedRole"
    />
  </div>
</template>

<script setup lang="ts">
import type { GameLobbyRolePickerDescriptionContentProps } from "~/components/pages/game-lobby/GameLobbyRolePicker/GameLobbyRolePickerDescription/GameLobbyRolePickerDescriptionContent/game-lobby-role-picker-description-content.types";
import GameLobbyRolePickerDescriptionBoundaries from "~/components/pages/game-lobby/GameLobbyRolePicker/GameLobbyRolePickerDescription/GameLobbyRolePickerDescriptionContent/GameLobbyRolePickerDescriptionBoundaries/GameLobbyRolePickerDescriptionBoundaries.vue";
import RoleDescriptionLines from "~/components/shared/role/RoleDescriptionLines/RoleDescriptionLines.vue";
import RoleOriginBadge from "~/components/shared/role/RoleOriginBadge/RoleOriginBadge.vue";
import RoleSideBadge from "~/components/shared/role/RoleSideBadge/RoleSideBadge.vue";
import RoleTypeBadge from "~/components/shared/role/RoleTypeBadge/RoleTypeBadge.vue";
import { useRoleName } from "~/composables/api/role/useRoleName";

const props = defineProps<GameLobbyRolePickerDescriptionContentProps>();

const { getRoleNameLabel } = useRoleName();

const roleName = computed(() => getRoleNameLabel(props.pickedRole.name));
</script>