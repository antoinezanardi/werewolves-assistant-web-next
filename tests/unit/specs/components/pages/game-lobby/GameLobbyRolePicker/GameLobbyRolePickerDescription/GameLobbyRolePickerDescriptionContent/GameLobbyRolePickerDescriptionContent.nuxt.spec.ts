import type { mount } from "@vue/test-utils";
import type { ComponentMountingOptions } from "@vue/test-utils/dist/mount";

import type { GameLobbyRolePickerDescriptionContentProps } from "~/components/pages/game-lobby/GameLobbyRolePicker/GameLobbyRolePickerDescription/GameLobbyRolePickerDescriptionContent/game-lobby-role-picker-description-content.types";
import GameLobbyRolePickerDescriptionContent from "~/components/pages/game-lobby/GameLobbyRolePicker/GameLobbyRolePickerDescription/GameLobbyRolePickerDescriptionContent/GameLobbyRolePickerDescriptionContent.vue";
import { createFakeRole } from "@tests/unit/utils/factories/composables/api/role/role.factory";
import { mountSuspendedComponent } from "@tests/unit/utils/helpers/mount.helpers";

describe("Game Lobby Role Picker Description Content Component", () => {
  let wrapper: ReturnType<typeof mount<typeof GameLobbyRolePickerDescriptionContent>>;
  const defaultProps: GameLobbyRolePickerDescriptionContentProps = {
    pickedRole: createFakeRole({
      name: "seer",
      type: "villager",
      side: "villagers",
      origin: "classic",
    }),
  };

  async function mountGameLobbyRolePickerDescriptionContentComponent(options: ComponentMountingOptions<typeof GameLobbyRolePickerDescriptionContent> = {}):
  Promise<ReturnType<typeof mount<typeof GameLobbyRolePickerDescriptionContent>>> {
    return mountSuspendedComponent(GameLobbyRolePickerDescriptionContent, {
      props: defaultProps,
      ...options,
    });
  }

  beforeEach(async() => {
    wrapper = await mountGameLobbyRolePickerDescriptionContentComponent();
  });

  it("should match snapshot when rendered.", () => {
    expect(wrapper).toBeTruthy();
    expect(wrapper.html()).toMatchSnapshot();
  });

  describe("Role Name", () => {
    it("should translate role name when rendered.", () => {
      const roleName = wrapper.find<HTMLHeadingElement>("#role-name");

      expect(roleName.text()).toBe("shared.role.name.seer");
    });
  });
});