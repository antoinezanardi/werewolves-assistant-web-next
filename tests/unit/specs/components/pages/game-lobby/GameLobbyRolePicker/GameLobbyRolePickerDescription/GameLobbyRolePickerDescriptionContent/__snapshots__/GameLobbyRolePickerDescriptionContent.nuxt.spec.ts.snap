// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Game Lobby Role Picker Description Content Component > should match snapshot when rendered. 1`] = `
"<div id="game-lobby-role-picker-description-content" class="flex flex-col gap-4 items-center w-full">
  <h3 id="role-name">shared.role.name.seer</h3>
  <role-side-badge-stub roleside="villagers" size="large" id="role-side-badge"></role-side-badge-stub>
  <div class="flex gap-2 items-center justify-between w-full">
    <role-origin-badge-stub roleorigin="classic" id="role-origin-badge"></role-origin-badge-stub>
    <role-type-badge-stub roletype="villager" id="role-type-badge" class="role-type-badge"></role-type-badge-stub>
  </div>
  <game-lobby-role-picker-description-boundaries-stub pickedrole="[object Object]" id="game-lobby-role-picker-description-boundaries" class="w-full"></game-lobby-role-picker-description-boundaries-stub>
  <divider-stub class="!my-2"></divider-stub>
  <role-description-lines-stub role="[object Object]" id="role-description-lines"></role-description-lines-stub>
</div>"
`;
